### DISCUSSION TITLE
Error: FW ID block not found after programmed

### DISCUSSION THREAD

**Discussion by RoyTien** [corundum/corundum#148] [Q&A] [answered]
Hello,
I have programmed a Xilinx U50 FPGA with Vivado.

By following the [Corundum Doc](https://docs.corundum.io/en/latest/gettingstarted.html#loading-the-fpga-design), I reset the FPGA after loaded the design.

![image](https://github.com/corundum/corundum/assets/11004295/1022afcb-5121-4170-85a8-b6cd39bc2718)

However, I QUERIED DEVICE INFO **FAILED** with `mqnic-fw`.

![image](https://github.com/corundum/corundum/assets/11004295/32a4dddb-30f4-466b-be7d-8f36b2bc7226)

The system is Linux version 4.15.0-112-generic (buildd@lcy01-amd64-027) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)). And I checked the code where the error info from.

![image](https://github.com/corundum/corundum/assets/11004295/9fbf9199-1a8d-43e3-b1f1-42fed0296fc9)
![image](https://github.com/corundum/corundum/assets/11004295/66eb0150-5d41-4327-b85d-7ad6377da2c2)

How can I fix this issue? @alexforencich
Thank you in advance for your time,
Kind regards,

**Comment by alexforencich** [comment]
Well, first of all you can't use mqnic-fw to talk to your PCIe root port (97:02.0), you need to point it at the FPGA (98:00.0).  Bur, you may also need to reboot after loading the firmware on the FPGA so the PCIe bus can be properly re-enumerated.  What does dmesg report while running the hot reset script?  And what does lspci -vvv -s 98:00.0 say?

**Comment by RoyTien** [comment]
Thank you so much @alexforencich, it works!!
I tried 98:00.0 and got the info successfully as shown. However, I got another "Error: Event queue block not found".

![image](https://github.com/corundum/corundum/assets/11004295/52a48aac-51a9-4e28-96eb-09bcb3ddef23)
![image](https://github.com/corundum/corundum/assets/11004295/d30e9c19-efd8-4317-93aa-c3c5aa55e252)

And the `dmesg` and `lspci -vvv -s 98:00.0` report:

![image](https://github.com/corundum/corundum/assets/11004295/40f75799-35d0-477c-9a9c-6432f296819b)
![image](https://github.com/corundum/corundum/assets/11004295/e50fc1c2-ba84-4cc0-a93b-d29a61a5b85d)

For now, should I cold reboot (`ipmitool power cycle`) to remove the event queue error?
Kind regards,

**Reply by alexforencich** [reply, marked as answer]
Event queue block not found implies a version conflict between the bitstream and the driver/userspace tools.  They both have to be built from the same git revision, otherwise you can get this sort of issue where the driver/userspace tools are looking for a particular register block with a particular version, but fail to find it.  Now, in this case, both the driver and the userspace tools don't completely fail to talk to the card, they just fail to set up/connect to the network interface related components, and this is specifically done to support reflashing the card via the driver/userspace tools despite some sort of version difference.  In your case, it may be as simple as make clean + make for both the userspace tools and the driver, but possibly you'll also have to rebuild the FPGA config.

**Reply by RoyTien** [reply]
Thank you so much, it is really helpful!
I printed out the device info without any error after I regenerated the bit file and loaded it into FPGA again.
Thank you again.
Kind regards,
